src/Val.h

```diff
@@ -506,38 +388,92 @@ class ValManager {
 
 extern ValManager* val_mgr;
 
+class IntVal : public Val {
+public:
+	IntVal(bro_int_t v)
+		: Val(base_type(TYPE_INT)), int_val(v)
+		{}
+
+	bro_int_t UnderlyingVal() const	{ return int_val; }
```

Comment:
I'm fine going with `Get()` and will include that change in my next update.  My only hesitation is that the name is a bit of an invitation to use the method rather than the `AsXYZ()` accessor.  We could also make the method protected so that externally only friends can use it.

Reply:
Why would using/exposing `Get()` be a problem? Actually I'm kind of hoping we could eventually go the opposite route and remove the `AsX` macros altogether and just do either explicit `dynamic_cast<XVal*>(v)->Get()` or some wrapper like `v->as<XVal>().Get()`.

Reply:
Ah okay now I see.  Makes sense.

Reply:
... but I'd like the conversion away from `AsX` to be separate from my current effort, as it requires a lot of code changes.

Reply:
Agree, would also breaks API more than necessary right now.